### PR TITLE
Improve countImpression functionality in native ios

### PR DESCRIPTION
**Pull request by slawomirzaba**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -157,9 +157,10 @@ Since we don't monitor the device's connection state you need to tell the SDK th
 contentPass.recoverFromError()
 ```
 
-### Couting an impression
-
-Counting an impression is as easy as calling the function `countImpression(completionHandler:)`. A user has to be authenticated and have an active subscription applicable to your scope for this to work.
+### Counting an impression
+`countImpression` method tracks and increments the impression count for the current user. This method should be invoked whenever
+a user views a piece of content. It applies to all users, whether authenticated or unauthenticated.
+Counting an impression is as easy as calling the function `countImpression(completionHandler:)`.
```
Comment (gerstnr) [inline README.md]:
```suggestion
### Counting an impression
`countImpression` method counts impressions for billing purposes. This method must be invoked whenever a user views a piece of content, independently of authentication state. If the current user is authenticated the impression will automatically be logged as paid ad-free impression to calculate the publisher compensation. As the total amount of impressions is required for billing as well, this method also counts sampled impressions of non-subscribers. Counting an impression is as easy as calling the function `countImpression(completionHandler:)`.
```

File: Sources/ContentPass/ContentPass.swift [hunk 2]
```diff
@@ -2,6 +2,8 @@ import AppAuth
 import AuthenticationServices
 import UIKit
 
+let samplingRate: Double = 0.05
```
Comment (0x7f) [inline Sources/ContentPass/ContentPass.swift:5]: Can we make this configurable via the config file? Also fine to do it in a separate PR if too much effort.

**Review by 0x7f — APPROVED** [review]

File: Sources/ContentPass/ContentPass.swift (hunk 2 above)
Comment (slawomirzaba) [reply Sources/ContentPass/ContentPass.swift:5]: it is, the question is if we need it for now?
